src/System.Collections.Immutable/tests/ImmutableArrayTest.cs

```diff
@@ -261,13 +350,13 @@ public void Covariance()
             ImmutableArray<string> derivedImmutable = ImmutableArray.Create("a", "b", "c");
             ImmutableArray<object> baseImmutable = derivedImmutable.As<object>();
             Assert.False(baseImmutable.IsDefault);
-            // Must cast to object or the IEnumerable<object> overload of Equals would be used
+            // Must cast to object or the IEnumerable<object> overload of Assert.Equal would be used
             Assert.Equal((object)derivedImmutable, baseImmutable, EqualityComparer<object>.Default);
 
             // Make sure we can reverse that, as a means to verify the underlying array is the same instance.
             ImmutableArray<string> derivedImmutable2 = baseImmutable.As<string>();
             Assert.False(derivedImmutable2.IsDefault);
-            Assert.Equal(derivedImmutable, derivedImmutable2);
+            Assert.True(derivedImmutable == derivedImmutable2);
```

Comment:
@ianhays When I first saw `Assert.Equal` I was confused because I thought it was calling the `Equal(IEnumerable<T>, IEnumerable<T>)` overload which compares the contents of the arrays. Hovering over the method in VS showed that this was not the case, but I still think this is better because the tests will fail loudly if you accidentally write `Assert.True(expected == source)` (where `source` is an enumerable and `expected` is an ImmutableArray) whereas they will silently pass if you write `Assert.Equal(expected, source)`.